src/kontrol/prove.py

```diff
@@ -413,7 +413,8 @@ def _method_to_cfg(
         new_node_ids = [init_node.id]
         init_node_id = init_node.id
 
-    is_test = method.signature.startswith('test')
+    proof_prefixes = ['test', 'proof', 'check']
+    is_test = any(method.signature.startswith(prefix) for prefix in proof_prefixes)
     failing = method.signature.startswith('testFail')
```

Comment:
Should we also have the dual logic for `proofFail` and `checkFail`?

Reply:
Yeah, I wasn't sure about that too, thanks @JuanCoRo! My reasoning behind not adding it was mostly the following:
- IIUC, the usage of `testFail` in Foundry itself is [generally discouraged](https://github.com/foundry-rs/foundry/issues/4437), considering that there is `vm.expectRevert()` cheatcode; it's a bit philosophical, but it also looks to me that proofs are not supposed to fail
- I don't think `hevm` and `Halmos` support the `...Fail` counterpart of these prefixes, so it not supporting them doesn't negatively affect compatibility with test suites written for these tools

But I agree they might be useful in some scenarios, so please let me know if you think I should add `proofFail` and `checkFail`, and I will!

Reply:
I don't know, is it common to use `proofFail`/`checkFail`?

Reply:
Well, I don't think anyone's supporting (or using) `proofFail` and `checkFail` per se, but maybe you're right @JuanCoRo and t makes sense to add those for consistency with `test` and `testFail`.